### PR TITLE
API: recursive tree request

### PR DESCRIPTION
**Pull request by rebecamendez**
Related pull request: https://github.com/gitlabhq/gitlabhq/pull/9185.
Params translated to boolean values.

I made a mistake in rebase so I prefer to start a clean pull request. You can close previous pull request.
Sorry for the inconvenience.

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -70,7 +70,7 @@ v 7.11.0 (unreleased)
   - Ensure that the first added admin performs repository imports
   - Update Asciidoctor gem to version 1.5.2. (Jakub Jirutka)
   - Fix resolving of relative links to repository files in AsciiDoc documents. (Jakub Jirutka)
-
+  - API: Allow recursive tree request
```
Comment (jvanbaarsen) [inline CHANGELOG]: Can you please move this to 7.12?

**Comment by rebecamendez** [comment]
Done

**Comment by rebecamendez** [comment]
Done.

**Comment by jvanbaarsen** [comment]
@rebecamendez Can you please make this mergeable again? Thanks :dancer:

**Comment by rebecamendez** [comment]
@jvanbaarsen How can I do that? with "git rebase master"? If I do that, my squashed commits looks split again. I'm new with the rebase feature :(

**Comment by dblessing** [comment]
@rebecamendez did you figure out how to rebase. Usually I will 

1. Change to master
2. Pull master from upstream
3. Change to branch
4. Run `git rebase master --preserve-merges`
5. Force push to branch.

**Comment by rebecamendez** [comment]
@dblessing Done! Thanks.

File: CHANGELOG [hunk 2]
```diff
@@ -60,6 +60,8 @@ v 7.14.1
   - Only include base URL in OmniAuth full_host parameter (Stan Hu)
   - Fix Error 500 in API when accessing a group that has an avatar (Stan Hu)
   - Ability to enable SSL verification for Webhooks
+  - Add FogBugz project import (Jared Szechy)
```
Comment (jvanbaarsen) [inline CHANGELOG]: Can you please make sure this is in the correct release point?

Comment (rebecamendez) [inline CHANGELOG]: @jvanbaarsen Done :)

**Comment by waxmihs2902** [comment]
thanks!

**Comment by connorshea** [comment]
Thanks for the contribution to GitLab! Unfortunately we're no longer accepting Pull Requests through the GitHub project.

If you'd like to get this merged, I encourage you to open this as a Merge Request on [the GitLab.com project](https://gitlab.com/gitlab-org/gitlab-ce). Feel free to ping me (@connorshea) and I'll get the right person assigned to it.

Apologies for the troubles, and thanks again for helping improve the project! :heart:

**Pull request closed without merging**